File: lib/models/user_model.dart
class UserModel {
  final int id;
  final String fullName;
  final String email;
  final String? bio;
  final List<String> interests;
  final List<String> skills;
  final bool hasExperience;
  final String? projectName;
  final String? experienceDescription;
  final String? goal;
  final bool hasOnboarded;

  UserModel({
    required this.id,
    required this.fullName,
    required this.email,
    this.bio,
    required this.interests,
    required this.skills,
    required this.hasExperience,
    this.projectName,
    this.experienceDescription,
    this.goal,
    required this.hasOnboarded,
  });

  factory UserModel.fromJson(Map<String, dynamic> json) {
    return UserModel(
      id: json['id'],
      fullName: json['full_name'],
      email: json['email'],
      bio: json['bio'],
      interests: List<String>.from(json['interests'] ?? []),
      skills: List<String>.from(json['skills'] ?? []),
      hasExperience: json['has_experience'] ?? false,
      projectName: json['project_name'],
      experienceDescription: json['experience_description'],
      goal: json['goal'],
      hasOnboarded: json['has_onboarded'] ?? false,
    );
  }
}

File: backend/models/user_model.py
from sqlalchemy import Column, Integer, String, Boolean, JSON
from database.connection import Base

class User(Base):
    __tablename__ = "users"

    id = Column(Integer, primary_key=True, index=True)
    full_name = Column(String(255))
    email = Column(String(255), unique=True)
    hashed_password = Column(String(255))

    # Onboarding Fields
    bio = Column(String(1000), nullable=True)
    interests = Column(JSON, nullable=True)
    skills = Column(JSON, nullable=True)
    has_experience = Column(Boolean, default=False)
    project_name = Column(String(255), nullable=True)
    experience_description = Column(String(2000), nullable=True)
    goal = Column(String(255), nullable=True)
    has_onboarded = Column(Boolean, default=False)